src/Capsule.cc

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright 2020 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+#include "sdf/Capsule.hh"
+
+using namespace sdf;
+
+// Private data class
+class sdf::CapsulePrivate
+{
+  // A capsule with a length of 1 meter and radius if 0.5 meters.
+  public: ignition::math::Capsuled capsule{1.0, 0.5};
+
+  /// \brief The SDF element pointer used during load.
+  public: sdf::ElementPtr sdf;
+};
+
+/////////////////////////////////////////////////
+Capsule::Capsule()
+  : dataPtr(new CapsulePrivate)
+{
+}
+
+/////////////////////////////////////////////////
+Capsule::~Capsule()
+{
+  delete this->dataPtr;
+  this->dataPtr = nullptr;
+}
+
+//////////////////////////////////////////////////
+Capsule::Capsule(const Capsule &_capsule)
+  : dataPtr(new CapsulePrivate)
+{
+  this->dataPtr->capsule = _capsule.dataPtr->capsule;
+  this->dataPtr->sdf = _capsule.dataPtr->sdf;
```

Comment:
nit: I recommend
```suggestion
  *dataPtr->capsule = *_capsule.dataPtr
```
since `CapsulePrivate` is copy assignable. That way we don't have to update this if we add more members to `CapsulePrivate`.

Reply:
feb90f6349db409fe440894f368ead987ee2ad4d